students/woogyeong/westagram/users/views.py

```diff
@@ -1 +1,33 @@
-from django.shortcuts import render
+# python built-in module
+import json
+
+# 외부 module
+from django.http  import JsonResponse
+from django.views import View
+
+# 사용자 module (직접 작성한 module)
+from users.models import User
+from users.validators import validate_email, validate_password
+
+class SignUpView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        
+        try:
+            if not validate_email(data['email']):
+                return JsonResponse({'message': 'Email format is not valid'}, status=400)
+            elif not validate_password(data['password']):
+                return JsonResponse({'message': 'Password format is not valid'}, status=400)
+            elif User.objects.filter(email=data['email']).exists():
+                return JsonResponse({'message': 'USER_EXISTS'}, status=400)
+            elif validate_email(data['email']) and validate_password(data['password']):
+                User.objects.create(
+                    email         = data['email'],
+                    mobile        = data['mobile'],
+                    user_name     = data['user_name'],
+                    user_id       = data['user_id'],
+                    password      = data['password'],
+                )
+                return JsonResponse({'message': "CREATED"}, status=201)
```

Comment:
```suggestion
            if not validate_email(data['email']):
                return JsonResponse({'message': 'Email format is not valid'}, status=400)
            
            if not validate_password(data['password']):
                return JsonResponse({'message': 'Password format is not valid'}, status=400)
            
            if User.objects.filter(email=data['email']).exists():
                return JsonResponse({'message': 'USER_EXISTS'}, status=400)
       
           User.objects.create(
                email         = data['email'],
                mobile        = data['mobile'],
                user_name     = data['user_name'],
                user_id       = data['user_id'],
                password      = data['password'],
            )
            return JsonResponse({'message': "CREATED"}, status=201)
```
1. if 조건을 만족하게 되면 return을 하므로 다시 elif로 안 묶어도 됩니다.
2. 위의 if 조건을 다 통과 했다는 것은 email, password가 조건을 다 만족한다는 의미이기 때문에 마지막 elif에 있던 조건을 없어도 될 것 같습니다.